### PR TITLE
ci: add testing, fix issues

### LINKED ISSUE #2: Add continuous integration based on azure-pipelines
See also https://github.com/scikit-build/scikit-build/pull/384

### PR DESCRIPTION
**Pull request by henryiii**
Swapping the custom code for testing to nox. We could easily parametrize later, such as over Python versions.

Adding the missing `pyproject.toml`'s to the rest of the projects. There seems to be some misbehaving with caching on scikit-build's part, hopefully won't affect CI, just reruns locally. ~~That's why I don't build wheels yet, as it forces the cache rebuild failure due to building twice immediately (and ideally, you would just install from the wheel instead of rebuilding again).~~ Patched the caching issues, though a proper fix should come from scikit-build eventually.

~~Currently cpp/cython fail~~, I think because they are not src based, so Python loads the local `hello` folder instead of the built one (I think). Edit: Yes, this is why. Moved the test file and worked around it for now.

Fixed a bunch of bugs in the tests, like missing `\n`'s and an extra `)`.

Closes #2.

### CODE REVIEW

**Comment by henryiii** [comment]
PS: Hoping to apply this to #8 after it's ready and merged.

**Comment by henryiii** [comment]
Comments for later PRs (not for now):

* Warnings are produced due to a couple of fields (author email and URL) missing.
* I'm not sure what the other two non-hello projects are supposed to do, so I just replicated the existing testing exactly there (sdist/wheel for pen2-cython, nothing for tower-of-babel).
* We should drop setup.py test. It's been deprecated for years.

**Comment by henryiii** [comment]
@jcfr or @thewtex (when you get out of the mountains :) ) , can I get a second set of eyes?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    branches:
+    - master
+
+jobs:
+  checks:
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-latest, macos-latest]
```
Comment (thewtex) [inline .github/workflows/ci.yml:15]: Can we add Windows? Or create an issue to add Windows later?

**Review by thewtex — APPROVED** [review]
@henryiii fantastic! :1st_place_medal: Clean CI is our ticket to progress. :rocket:

**Comment by thewtex** [comment]
> I'm not sure what the other two non-hello projects are supposed to do, so I just replicated the existing testing exactly there (sdist/wheel for pen2-cython, nothing for tower-of-babel).

I think these help demonstrate generating a command line executable based on cython and a non-trivial package that is cython + python. We may want to re-name them to identify them as such.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (henryiii) [reply .github/workflows/ci.yml:15]: scikit-build has issues with Windows, see https://github.com/scikit-build/scikit-build/pull/526 . Eventually, yes.

Comment (henryiii) [reply .github/workflows/ci.yml:15]: cpp and pure work on Windows, the others didn't. Can try adding it back after this goes in.

**Pull request merged**